chajda/tsquery/augments.py

```diff
@@ -52,62 +52,185 @@ def augments_gensim(lang, word, config=Config(), n=5):
 
     return words
 
-
+import os
 import fasttext
 import fasttext.util
+from annoy import AnnoyIndex
+
+
+chajda_path = os.path.dirname(__file__)
+working_dir = os.getcwd()
 
 fasttext_models = {}
+annoy_indices = {}
+
+
+def create_annoy_index_if_needed(lang, word='google', n=5):
+    '''
+    Checks whether an AnnoyIndex has already been created for a specified language.
+    Populates an AnnoyIndex  with vectors from the fasttext model that corresponds to language,
+    builds the index, saves it to disk. 
+    The AnnoyIndex is stored in the annoy_indices dictionary.
 
-def augments_fasttext(lang, word, config=Config(), n=5):
+    lang: language passed from augments_fasttext
+    
+    NOTE:
+    Populating an AnnoyIndex with vectors from a fasttext model requires more memory than github actions allows for.
+    Therefore, the doctest has been commented out.
+    # >>> create_annoy_index_if_needed('en') or annoy_indices['en'] != None
+    # True
+    '''
+    # create AnnoyIndex if not already created
+    try:
+        annoy_indices[lang]
+    except KeyError:
+        annoy_indices[lang] = AnnoyIndex(300, 'angular')
+
+        # if annoy index has not been saved for this language yet,
+        # populate annoy index with vectors from corresponding fasttext model
+        try:
+            annoy_indices[lang].load(f"{chajda_path}/annoy/{lang}.ann")
+
+        # OSError occurs if index is not already saved to disk
+        except OSError:
+            # The annoy library uses a random number genertor when building up the trees for an Annoy Index.
+            # In order to ensure that the output is deterministic, we specify the seed value for the random number generator.
+            annoy_indices[lang].set_seed(22)
+
+            # Populate AnnoyIndex with vectors from corresponding fasttext model
+            i = 0
+            for j in fasttext_models[lang].words:
+                v = fasttext_models[lang][j]
+                annoy_indices[lang].add_item(i,v)
+                i += 1
+
+            # build the trees for the index    
+            annoy_indices[lang].build(10)
+
+            # save the index to annoy directory,
+            # make the directory if needed
+            try:
+                annoy_indices[lang].save(f"{chajda_path}/annoy/{lang}.ann")
+            except OSError:
+                os.mkdir(f"{chajda_path}/annoy")
+                annoy_indices[lang].save(f"{chajda_path}/annoy/{lang}.ann")
+
+
+def load_fasttext_model(lang, word='google', n=5):
+    '''
+    Downloads and loads a fasttext model for a particular language,
+    storing the fasttext model object in the fasttext_models dictionary.
+    Due to the size of fasttext models (>1gb), and the loading time,
+    we do not load all models on program startup.
+    Instead, this function is used to load them lazily as needed. 
+
+    lang: language passed from augments_fasttext
+    
+    # >>> load_fasttext_model('en') or fasttext_models['en'] != None
```

Comment:
why is this commented?

Reply:
I just read the note that you had for the annoy function; you should have the same note here if that's the reason